src/urllib3/contrib/emscripten/fetch.py

```diff
@@ -0,0 +1,477 @@
+"""
+Support for streaming http requests in emscripten.
+
+A few caveats -
+
+Firstly, you can't do streaming http in the main UI thread, because atomics.wait isn't allowed.
+Streaming only works if you're running pyodide in a web worker.
+
+Secondly, this uses an extra web worker and SharedArrayBuffer to do the asynchronous fetch
+operation, so it requires that you have crossOriginIsolation enabled, by serving over https
+(or from localhost) with the two headers below set:
+
+    Cross-Origin-Opener-Policy: same-origin
+    Cross-Origin-Embedder-Policy: require-corp
+
+You can tell if cross origin isolation is successfully enabled by looking at the global crossOriginIsolated variable in
+javascript console. If it isn't, streaming requests will fallback to XMLHttpRequest, i.e. getting the whole
+request into a buffer and then returning it. it shows a warning in the javascript console in this case.
+
+Finally, the webworker which does the streaming fetch is created on initial import, but will only be started once
+control is returned to javascript. Call `await wait_for_streaming_ready()` to wait for streaming fetch.
+"""
+from __future__ import annotations
+
+import io
+import json
+from email.parser import Parser
+from typing import TYPE_CHECKING, Any
+
+import js  # type: ignore[import]
+from pyodide.ffi import JsArray, JsException, JsProxy, to_js  # type: ignore[import]
+
+if TYPE_CHECKING:
+    from typing_extensions import Buffer
+
+from .request import EmscriptenRequest
+from .response import EmscriptenResponse
+
+"""
```

Comment:
Instead of always ignoring `User-Agent`, we should document that we don't send our default `User-Agent` when using emscripten however we still want to respect users preferences if /they/ want to send a `User-Agent`, in that case we document the potential downsides of doing so.

Reply:
I think this is a bad idea - it breaks firefox if anyone uses a library that sets its own user agent. `requests` being the obvious one. It means that that library will just flat-out not work on e.g. github pages.

Reply:
Maybe an emscripten specific setting api to enable user-agent changes or something.